digester.go

```diff
@@ -62,50 +63,65 @@ func (t *tarFilterer) Write(p []byte) (int, error) {
 }
 
 func (t *tarFilterer) Close() error {
+	t.closed = true
```

Comment:
Should you check if t.closed is true, and return success or an error, rather then executing it a second time?

Reply:
Can't hurt.  Added a check there.

Reply:
Good news: we're definitely running the unit tests.  Bad news: they found a race.  Good news: added a lock to fix it.